CI fails with "Problem creating zip: Execution exception: Java heap space"

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

Some the CI builds fail with 

> Problem creating zip: Execution exception: Java heap space"

```
[INFO] Keycloak Quarkus Server App ........................ SUCCESS [  8.215 s]
[INFO] Keycloak Quarkus Server Distribution ............... FAILURE [  4.968 s]
[INFO] Keycloak Quarkus Test Parent ....................... SKIPPED
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SKIPPED
[INFO] Keycloak Quarkus Server Integration tests .......... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:45 min
[INFO] Finished at: 2025-02-04T08:00:23Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.1.1:single (assemble) on project keycloak-quarkus-dist: Failed to create assembly: Error creating assembly archive server-dist: Problem creating zip: Execution exception: Java heap space -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :keycloak-quarkus-dist
Error: Process completed with exit code 1.
```


### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

The build should succeed

### Actual behavior

It fails

### How to Reproduce?

https://github.com/keycloak/keycloak/actions/runs/13130744495/job/36635376079?pr=37007

Seems to happen only sometimes.

### Anything else?

I'll create a PR to increase the heap size from 1 MB to 1.5 MB

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -281,7 +281,7 @@ jobs:
       fail-fast: false
     runs-on: ${{ matrix.os }}
     env:
-      MAVEN_OPTS: -Xmx1024m
+      MAVEN_OPTS: -Xmx1536m
     steps:
       - uses: actions/checkout@v4
 
@@ -785,7 +785,7 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 35
     env:
-      MAVEN_OPTS: -Xmx1024m
+      MAVEN_OPTS: -Xmx1536m
     steps:
       - uses: actions/checkout@v4
 

